### ISSUE TITLE
Replace methods with arrow class properties

### ISSUE THREAD

**Issue by micha149** [dmtrKovalenko/date-io#84]
Im am using `date-io`in combination with `@material-ui/pickers` and `@date-io/date-fns`.  I came across some error which is caused by the class based implementation of https://github.com/dmtrKovalenko/date-io/blob/master/packages/date-fns/src/date-fns-utils.ts

Here is what I was trying to do:
```javascript
import { useUtils } from '@material-ui/pickers';

const MyComponent = () => {
    const { isSameDay, isValid } = useUtils();

    isValid(someDate);
}
```

The error occured in the `isValid` call, because it uses internally `this.date()` and I suspect that `this` is not the right context when the method was destructured.

Maybe that this usage is not what you intended. But working with a factory function could lead to a better developer expierience.

Example:
```javascript
// ...

const ensureDate = date(value?: any) {
    if (typeof value === "undefined") {
        return new Date();
    }

    if (value === null) {
        return null;
    }

    return new Date(value);
}

// ...

export default function createDateFnsAdapter(locale): IUtils<Date> {
    // ...
    const getDatePickerHeaderText = (date: Date) => format(date, "EEE, MMM d", { locale });
    const isValid = (value: any) => isValid(ensureDate(value));
    // ...

    return {
        // ...
        getDatePickerHeaderText,
        isValid,
        // ...
    }
}
```

**Comment by dmtrKovalenko** [comment]
I know that, but class style also allowes more expressive override style, so we will not get rid of it. 

But I think it’s possible to make all functions in class to be written as class properties. So they will not lose context. Feel free to open a PR

**Comment by micha149** [comment]
Isn't it expressive enough to use factory functions and rest operators?

```javascript
export default function createDateFnsAdapter(locale): IUtils<Date> {
    const baseImplementation = createMomentJsAdapter(locale);
    
    const isValid = (value: any) => isValid(ensureDate(value));
    // ...

    return {
        ...baseImplementation,
        isValid,
        // ...
    }
}
```

Did you mean to assign arrow function to class properties? Isn't it still a proposal? Another solution is to bind all methods in the constructor. But this is extreme ugly…

**Comment by dmtrKovalenko** [comment]
Refactoring to the `() => {}` arrow functions in the class would be perfect.

**Comment by dmtrKovalenko** [comment]
Closed by #483
